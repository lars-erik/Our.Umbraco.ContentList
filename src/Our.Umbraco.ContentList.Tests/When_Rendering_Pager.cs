using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using Microsoft.AspNetCore.Mvc.Razor;
using Microsoft.AspNetCore.Mvc.ViewFeatures;
using NUnit.Framework;
using Our.Umbraco.ContentList.Models;
using Our.Umbraco.ContentList.Tests.Support;

namespace Our.Umbraco.ContentList.Tests
{
    [TestFixture]
    public class When_Rendering_Pager
    {
        private TestServerFactory serverFactory;

        [Test]
        public async Task On_First_Page_Renders_One_Through_Five()
        {
            serverFactory = new TestServerFactory();
            var viewRenderer = serverFactory.GetRequiredService<ViewRenderer>();

            var pagerModel = new ContentListPaging
            {
                Total = 100,
                From = 0,
                To = 9,
                Page = 1,
                PageSize = 10,
                ShowPaging = true
            };
            
            var result = await viewRenderer.Render("PagerTest", pagerModel);
            
            Console.WriteLine(result);
            Approvals.VerifyHtml(result);
        }
    }
}
