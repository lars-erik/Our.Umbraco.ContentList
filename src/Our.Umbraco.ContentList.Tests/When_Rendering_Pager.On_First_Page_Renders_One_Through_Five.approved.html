<nav aria-label="Page navigation">
  <ul class="pagination">
    <li class="page-item">1</li>
    <li class="page-item">
      <a class="page-link" href="#">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="#">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="#">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="#">5</a>
    </li>
  </ul>
</nav>